### PR TITLE
[feat] GRU 모델 초기 세팅

### LINKED ISSUE #4: 주가 예측 GRU 모델 초기 세팅

### PR DESCRIPTION
**Pull request by byb-8**
## 개요
주가 예측 GRU 모델 초기 세팅
## 변경사항
dataloader를 통해 데이터 다운로드와 전처리를 하였고, model에선 GRU 모델을 구성하였다.
그리고 trainer에선 모델의 학습과 테스트를 하였고, main에선 하이퍼 파라미터를 세팅하였다.

### CODE REVIEW

File: trainer.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        loss.backward()
+        optimizer.step()
+
+        print("Epoch: ", epoch + 1, "Loss:", loss.item())
+
+def test(model, x_test, y_test, criterion, df):
+    model.eval()
+    with torch.no_grad():
+        outputs = model(x_test)
+        loss = criterion(outputs, y_test)
```
Comment (kdyeon0309) [inline trainer.py]: 사용하지 않는 loss 변수는 지워주는게 맞는 것 같습니다.

File: main.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    input_size = 4
+    hidden_size = 2
+    num_layers = 1
+    num_classes = 1
+    model=GRU(num_classes, input_size, hidden_size, num_layers, x_train.shape[1])
+    criterion = nn.MSELoss()
+    optimizer = torch.optim.Adam(model.parameters(), lr=0.0001)
+    train(model, x_train, y_train, criterion, optimizer, 10000)
+    test(model, x_test, y_test, criterion, df)
+
```
Comment (kdyeon0309) [inline main.py]: 학습을 완료하고 해당 모델을 왜 저장하지 않나요

File: dataloader.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    y=df[['Close']]
+    # print(x.shape)
+    ms=MinMaxScaler()
+    ss=StandardScaler()
+    x=ss.fit_transform(x)
+    y=ms.fit_transform(y)
+    x_train=x[:74, :]
+    x_test=x[74:, :]
+    y_train=y[:74, :]
+    y_test=y[74:, :]
```
Comment (kdyeon0309) [inline dataloader.py]: 현재 불러온 데이터셋은 네이버의 2023-06-01부터 2024-06-30까지의 주가 데이터로 총 264개의 데이터로 이루어져있습니다.
지금 train data를 74개 test 데이터를 190개로 test data의 비율이 train data보다 2배 이상 많은데 왜 이렇게 데이터를 나누셨는지 궁금합니다.

File: main.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    x_train, y_train, x_test, y_test=get_dataset()
+    x_train, y_train, x_test, y_test, df=get_dataset()
+    input_size = 4
+    hidden_size = 2
+    num_layers = 1
+    num_classes = 1
+    model=GRU(num_classes, input_size, hidden_size, num_layers, x_train.shape[1])
+    criterion = nn.MSELoss()
+    optimizer = torch.optim.Adam(model.parameters(), lr=0.0001)
+    train(model, x_train, y_train, criterion, optimizer, 10000)
```
Comment (kdyeon0309) [inline main.py]: 지금 현재 train 데이터는 74개 정도로 매우 적은 편입니다 하지만 epoch의 값은 10000으로 엄청나게 높은 편인데 이렇게 epoch값을 높게 준다면 overfiting 문제가 일어나지 않나요?

File: main.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    test(model, x_test, y_test, criterion, df)
+
+if __name__=='__main__':
+    main()
```
Comment (kdyeon0309) [inline main.py]: 각각의 하이퍼파라미터들을 저장하여 여러가지 테스트해 최적의 파라미터를 찾아내는 파라미터 튜닝 코드가 필요할 것 같습니다

File: dataloader.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import yfinance as yf
```
Comment (kdyeon0309) [inline dataloader.py]: 현재 저희는 주가예측 모델과 주식시장 호재 분석 모델로 총 2가지의 모델을 개발할 예정이니 각 모델 별 디렉토리를 나눠주면 좋을 것 같습니다.

File: trainer.py (hunk 1 above)
Comment (byb-8) [reply trainer.py]: [25106c3](https://github.com/Jusicool-Ver-2-0/Jusicool-Ai/pull/5/commits/25106c3bee5fe4b7febf12a7d57c4d60463ad6ea)
수정하였습니다

File: main.py (hunk 2 above)
Comment (byb-8) [reply main.py]: 제 실수네요.....
추후에 추가하겠습니다...

File: dataloader.py (hunk 3 above)
Comment (byb-8) [reply dataloader.py]: 제가 생각없이 이렇게 나눈 것 같습니다. 현재 한 종목을 가지고 train data와 test data를 나누고 있는데, 추후에 한 종목으로 train과 validation data를 나누고, 다른 한 종목 전체를 test 데이터로 쓰는 것을 고려해보고 있는데 이 부분에 대해 어떻게 생각하시는 지 궁금합니다.

File: main.py (hunk 4 above)
Comment (byb-8) [reply main.py]: 넵 저도 overfiting 문제가 일어날 것 같다고 생각합니다... 그래서 추후에 validation 부분을 추가해 overfiting이 일어나는 지 확인하고 해결하겠습니다

File: main.py (hunk 5 above)
Comment (byb-8) [reply main.py]: 넵 추후에 추가하겠습니다

File: dataloader.py (hunk 6 above)
Comment (byb-8) [reply dataloader.py]: [5edc1bd](https://github.com/Jusicool-Ver-2-0/Jusicool-Ai/pull/5/commits/5edc1bd4721ee048b2def256191c62c6812e2f16)
추가했습니다
